app/controllers/pieces_controller.rb

```diff
@@ -17,9 +17,15 @@ def update
 
     pos_x = params[:pos_x].to_i
     pos_y = params[:pos_y].to_i
+    valid_move = false
 
     Piece.transaction do
-      @piece.do_move!(pos_x, pos_y)
+      if @piece.valid_move?([pos_x, pos_y])
+        valid_move = true
+        @piece.do_move!(pos_x, pos_y)
+      else
+        valid_move = false
+      end
   	end
```

Comment:
I know, spacing looks horrible. I made my files indent with 2 spaces, but somehow that seems to make it worse when pushed up... :/